C-Plus-Plus/graphs/BellmanFord.cpp

```diff
@@ -0,0 +1,72 @@
+
+/*Input:
+First line contains two space separated integers,(N,M) 
+N- no of vertices, M- no of edges.
+Then M lines follow, each line has 3 space separated integers ui ,vi ,wi which denotes edge from vertex ui to vi with weight wi.
+Find shortest distance from vertex 1 to all other vertices.
+
+
+TEST CASE
+5 5
+1 2 5
+1 3 2
+3 4 1
+1 4 6
+3 5 5
+
+OUTPUT
+5 2 3 7 */
+
+
+#include<bits/stdc++.h>
+using namespace std;
+
+vector<int>edges[1000001];
+int dist[1000001];
+
+void bellmanFord(int n){
+  for(int i=0;i<n-1;i++){
+    int k=0;
+    while(edges[k].size()>0){
+      int u=edges[k][1],v=edges[k][2],w=edges[k][0];
+      if(dist[v]>dist[u]+w){
+        dist[v]=dist[u]+w;
+      }
+      k++;
+    }
```

Comment:
@Sidhant-Gupta Though you have added separate code for`DetectNegativeCycle.cpp` but here also add code how to deal with negative cycle to make it more robust. You can iterate the inner loop once more to check for the negative cycle and just the output `Graph contains the negative weight cycle. Hence, the shortest distance not guaranteed.`.

Reply:
Thank you for the suggestions @iamrajiv.  I've updated the code accordingly.